### PR TITLE
Externalize HS config inside the app configuration file

### LINKED ISSUE #58: Externalize HS config inside the config.json
Externalize HS config inside the config.json

### PR DESCRIPTION
**Pull request by Yoshiin**

### CODE REVIEW

File: src/components/structures/login/Login.js [hunk 1]
```diff
@@ -64,12 +63,15 @@ module.exports = React.createClass({
...
+        const randomHs = TCHAP_HOSTS_BASE + tchapHostsList[(Math.floor(Math.random() * (tchapHostsList.length)) + 1) - 1];
+
         return {
             busy: false,
             errorText: null,
             loginIncorrect: false,
-            enteredHomeserverUrl: this.props.customHsUrl || this.props.defaultHsUrl,
-            enteredIdentityServerUrl: this.props.customIsUrl || this.props.defaultIsUrl,
+            enteredHomeserverUrl: randomHs || this.props.customHsUrl || this.props.defaultHsUrl,
+            enteredIdentityServerUrl: randomHs || this.props.customIsUrl || this.props.defaultIsUrl,
```
Comment (dbkr) [inline src/components/structures/login/Login.js]: I'm not really sure what the intention is here:

 * Isn't the login HS supposed to be discovered by querying a random *IS* (in `discoverTchapPlatform`) and then using the resulting HS?
 * Shouldn't the custom server URL take precedence if it's been entered?
 * Doesn't this make the default servers redundant?

File: src/components/views/directory/NetworkDropdown.js [hunk 2]
```diff
@@ -154,7 +141,11 @@ export default class NetworkDropdown extends React.Component {
             servers.unshift(MatrixClientPeg.getHomeServerName());
         }
 
-        SERVER_LIST.forEach(s => {
+        let hsMainList = SdkConfig.get()['hs_main_list'];
+        let svrList = hsMainList.concat(SdkConfig.get()['hs_additional_list']);
+        svrList.push('all.tchap.gouv.fr');
```
Comment (dbkr) [inline src/components/views/directory/NetworkDropdown.js]: I'm not sure what this is?

Comment (dbkr) [reply src/components/views/directory/NetworkDropdown.js]: never mind - apparently it's a thing to get results from all HSes

File: src/components/structures/RoomDirectory.js [hunk 3]
```diff
@@ -103,7 +90,8 @@ module.exports = React.createClass({
             });
         });
 
-        let serverList = this.state.serverList;
+        let hsMainList = SdkConfig.get()['hs_main_list'];
+        let serverList = hsMainList.concat(SdkConfig.get()['hs_additional_list']);
```
[the remark this answers is not in the document]
Comment (Yoshiin) [reply src/components/structures/RoomDirectory.js]: I can verify if the vars aren't null, but, if the user reached this point, 'hs_main_list' should be defined.
Do you think I have to check the vars and spawn an error if one of them are not defined ?

[the remark this answers is not in the document]
Comment (dbkr) [reply src/components/structures/RoomDirectory.js]: Well if there is no `hs_main_list` in the config, won't `hsMainList` be `undefined` when this line is run, causing an exception?

File: src/components/structures/login/Login.js (hunk 1 above)
Comment (dbkr) [reply src/components/structures/login/Login.js]: OK - if you don't want the custom server stuff it probably makes sense to remove it altogether?

File: src/components/structures/login/Login.js [hunk 4]
```diff
@@ -64,12 +63,15 @@ module.exports = React.createClass({
     },
 
     getInitialState: function() {
+        let tchapHostsList  = SdkConfig.get()['hs_main_list'];
+        const randomHs = TCHAP_HOSTS_BASE + tchapHostsList[(Math.floor(Math.random() * (tchapHostsList.length)) + 1) - 1];
```
Comment (dbkr) [inline src/components/structures/login/Login.js]: This will still throw if there's no `hs_main_list` in the config file

**Comment by Yoshiin** [comment]
> Also I think something has gone wrong with the git branching: there's a force-push in here now and a load of unrelated changes have been pulled in.

Not really. this is due to the conflict resolution with develop. If you speak about https://github.com/dinsic-pim/matrix-react-sdk/blob/4303fd12589492623f561f385c2b0f432476a564/src/components/structures/RoomDirectory.js#L425  this was present in the initial commit.

**Comment by Yoshiin** [comment]
Closed due to a lot of confusion from conflict resolution.
New PR : #88

**Pull request closed without merging**